src/helpers/post-helper.php

```diff
@@ -164,6 +164,11 @@ public function update_has_public_posts_on_attachments( $post_parent, $has_publi
 	 * @return bool True if the post can be indexed.
 	 */
 	public function is_post_indexable( $post_id ) {
+		// Don't index posts which are not public (i.e. viewable).
+		if ( ! \is_post_type_viewable( \get_post_type( $post_id ) ) ) {
```

Comment:
By adding this, we now show "yoast_head":null and "yoast_head_json":null in the /wp-json/wp/v2/CPT REST API endpoint of a CPT that is not Public Queryable whereas we used to populate those. As per [this convo](https://yoast.slack.com/archives/C03KU0EHCNQ/p1666347057759259), we at least need to we make those'empty' (strings) in the `Yoast_Head_REST_Field` probably. Then existing stuff would fail more gracefully.

Reply:
Also, by adding this, now the indexable posts from the builder are different from those queried in the `Indexable_Post_Indexation_Action` and I am not sure how we can unify them, so lets go back to the drawing board for that one maybe?

Reply:
The same things can be said for the terms as well